pilot/proxy/envoy/policy.go

```diff
@@ -120,3 +125,128 @@ func applyClusterPolicy(cluster *Cluster,
 		}
 	}
 }
+
+func applyLoadBalancePolicy(cluster *Cluster, policy *routingv2.LoadBalancerSettings) {
+	if policy == nil || cluster.Type == ClusterTypeOriginalDST {
+		return
+	}
+
+	if consistent := policy.GetConsistentHash(); consistent != nil {
+		// cluster.LbType = LbTypeRingHash
+		// TODO: need to set special values in the route config
+		// see: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/load_balancing.html#ring-hash
+		log.Warn("Consistent hash load balancing type is not currently supported")
+	} else {
+		switch policy.GetSimple() {
+		case routingv2.LoadBalancerSettings_LEAST_CONN:
+			cluster.LbType = LbTypeLeastRequest
+		case routingv2.LoadBalancerSettings_RANDOM:
+			cluster.LbType = LbTypeRandom
+		case routingv2.LoadBalancerSettings_PASSTHROUGH:
+			cluster.LbType = LbTypeOriginalDST // FIXME: double-check
+		default:
+			cluster.LbType = LbTypeRoundRobin
+		}
+	}
+}
+
+func buildOutlierDetection(outlier *routingv2.OutlierDetection) *OutlierDetection {
+	if outlier != nil && outlier.Http != nil {
+		out := &OutlierDetection{
+			ConsecutiveErrors:  5,
+			IntervalMS:         10000, // 10s
+			BaseEjectionTimeMS: 30000, // 30s
+			MaxEjectionPercent: 10,
+		}
+
+		if outlier.Http.BaseEjectionTime != nil {
+			out.BaseEjectionTimeMS = protoDurationToMS(outlier.Http.BaseEjectionTime)
+		}
+		if outlier.Http.ConsecutiveErrors > 0 {
+			out.ConsecutiveErrors = int(outlier.Http.ConsecutiveErrors)
+		}
+		if outlier.Http.Interval != nil {
+			out.IntervalMS = protoDurationToMS(outlier.Http.Interval)
+		}
+		if outlier.Http.MaxEjectionPercent > 0 {
+			out.MaxEjectionPercent = int(outlier.Http.MaxEjectionPercent)
+		}
+
+		return out
+	}
+	return nil
+}
+
+func applyConnectionPool(cluster *Cluster, settings *routingv2.ConnectionPoolSettings) {
+	cluster.MaxRequestsPerConnection = 1024 // TODO: set during cluster construction?
```

Comment:
This is Envoy's default, right? If we don't set it at all will it do this for us?

Reply:
Yep. We explicitly set it to the Istio default to make the code clearer, even though it is technically non-necessary unless the Istio default was different than the Envoy default.